### ISSUE TITLE
Baichuan2-7b-chat cannot output the </s> eos token.

### ISSUE THREAD

**Issue by ouwenjie03** [vllm-project/vllm#1561]
My test code is `example/offline_inference.py`, the input prompt is: `"<reserved_106>How are you?<reserved_107>"`.
The output of baichuan2-7b-chat is 
```
I am fine thank you :) How about yourself? How are you doing today/now? How's everything going/going well/going well/going well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing well/doing
```

But the output of baichuan2-13b-chat is OK:
```
I'm doing well, thank you for asking. How about you?
```

I am sure that 7b model uses the ROPE and 13B model uses the ALIBI. 
The previous part of output looks correct. The problem seems like cannot output the eos token.

Does anyone meet the same problem?

**Comment by ouwenjie03** [comment]
I found something different between baichuan1 and baichuan2.
baichuan2 use the `NormHead` for the `lm_head`, so in the `load_weights`, we need to normalize the `loaded_weight`:
```python
if 'lm_head' in name:
    loaded_weight = nn.functional.normalize(loaded_weight)
```
Then everything goes well.

**Comment by liulfy** [comment]
how to run Baichuan2-13b-chat and Baichuan2-7b-chat? Could you please share your code? Thanks~

**Comment by RetormLi** [comment]
> I am sure that 7b model uses the ROPE and 13B model uses the ALIBI.
Thank you for mentioning this. I found a very misleading trap between Baichuan and Baichuan2.
Vllm will distinguish models by their "architecture" defined in config.json. Baichuan-7b is "BaiChuanForCausalLM" and Baichuan-13b is "BaichuanForCausalLM" (different capital letter). So Vllm will detect that and apply a different positional embedding.
But for Baichuan2, both Baichuan2-7b and Baichuan2-13b is "BaichuanForCausalLM". So Vllm will directly deploy a ALIBI positional embedding for both of them, which is not correct for Baichuan2-7b (right one is ROPE).
So the corrent way to use Baichuan2-7b is:
1. merge lora model if needed
2. do normalization to the saved "lm_head" weights
3. change the "architecture" to "BaiChuanForCausalLM"
